sdk/recoveryservices/azure-mgmt-recoveryservicesbackup/azure/mgmt/recoveryservicesbackup/activestamp/_patch.py

```diff
@@ -24,8 +24,42 @@
 # IN THE SOFTWARE.
 #
 # --------------------------------------------------------------------------
+import copy
+from typing import List
+import importlib
+import urllib.parse
+from ._recovery_services_backup_client import RecoveryServicesBackupClient as RecoveryServicesBackupClientGenerated
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+class RemoveDuplicateParamsPolicy(SansIOHTTPPolicy):
+    def __init__(self, duplicate_param_names):
+        # type: (List[str]) -> None
+        self.duplicate_param_names = duplicate_param_names
+
+    def on_request(self, request):
+        parsed_url = urllib.parse.urlparse(request.http_request.url)
+        query_params = urllib.parse.parse_qs(parsed_url.query)
+        updated_query_params = copy.copy(query_params)
+        for query in query_params:
+            if query in self.duplicate_param_names and (isinstance(query_params[query], list) and len(query_params[query]) > 1):
+                updated_query_params[query] = [query_params[query][-1]]
```

Comment:
Do we want the first or the last value? This grabs the last value. And when would you not get a `list` as the value?

Python nitpick:
You may consider using dict comprehension for this:

```python
{ key: val[-1:] if key in self.duplicate_names else val for key, val in query_params.items()}
``` 

...although it is arguably not easier to understand compared to what you have now....

Reply:
the last value. urllib parse returns the queries in the order they are in the url. The way we end up formatting, we format at the beginning of the query, so the last value is going to be the one the service returned.

Yeah let me try with python dict comprehension, i thought i'd be ok with doing copy to keep it cleaner but it's bothering me more and more. thanks!

Reply:
updated!